python/ray/worker.py

```diff
@@ -2078,7 +2104,9 @@ def connect(info,
     assert worker.cached_remote_functions_and_actors is not None, error_message
     # Initialize some fields.
     worker.worker_id = random_string()
-
+    if temp_dir is not None:
+        # Override the temporary directory.
+        tempfile_services.set_temp_root(temp_dir)
```

Comment:
the only time we should call this is in `start_ray_processes`

Reply:
maybe default_worker.py also

Reply:
I agree. @suquark could you fix this?